### PR TITLE
Remove test forward compat

### PR DESCRIPTION
**Pull request by Kordyjan**
Fixes https://github.com/lampepfl/dotty/issues/14989

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -363,6 +363,7 @@ jobs:
           git submodule update --init --recursive --jobs 7
           ./project/scripts/sbt "community-build/testOnly dotty.communitybuild.CommunityBuildTestC"
 
+<<<<<<< HEAD
```
Comment (nicolasstucki) [inline .github/workflows/ci.yaml:366]: You have some unresolved merge conflicts

Comment (Kordyjan) [reply .github/workflows/ci.yaml:366]: Locally I don't have those conflicts. Fetching / pulling doesn't change anything. Am I going insane?

Comment (Kordyjan) [reply .github/workflows/ci.yaml:366]: I'll check that on the fresh repo clone.

Comment (Kordyjan) [reply .github/workflows/ci.yaml]: Should be better now.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -509,7 +462,7 @@ jobs:
         - ${{ github.workspace }}/../../cache/sbt:/root/.sbt
         - ${{ github.workspace }}/../../cache/ivy:/root/.ivy2/cache
         - ${{ github.workspace }}/../../cache/general:/root/.cache
-    needs: [test_non_bootstrapped, test, mima, community_build_a, community_build_b, community_build_c, community_build_forward_compat, test_sbt, test_java8]
```
Comment (nicolasstucki) [inline .github/workflows/ci.yaml:512]: Why was `mima` removed?

Comment (Kordyjan) [reply .github/workflows/ci.yaml:512]: `git revert` shenanigans. I didn't touch this line.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -363,53 +363,6 @@ jobs:
           git submodule update --init --recursive --jobs 7
           ./project/scripts/sbt "community-build/testOnly dotty.communitybuild.CommunityBuildTestC"
 
-      - name: Show dependency tracking file
```
Comment (nicolasstucki) [inline .github/workflows/ci.yaml:366]: Did you remove this on purpose? Or is it another git revert issue

Comment (Kordyjan) [reply .github/workflows/ci.yaml]: Definitely not intentional.

**Comment by Kordyjan** [comment]
@smarter Do we need to backport that into 3.1.3? I don't see any reason.

**Comment by smarter** [comment]
I was thinking that if we're planning to definitely remove scala-output-release, we should remove it from 3.1.3 so that people don't start relying on it.

**Comment by Kordyjan** [comment]
This PR only removes forward compatibility tests from the community build. They are triggered only on nightly builds, so they don't matter outside of the `main` branch.

Actual removing the feature requires more caution and will come as a PR next week.

**Comment by smarter** [comment]
Ah I see, thanks!

**Review by romanowski — APPROVED** [review]

**Pull request merged**